### DISCUSSION TITLE
Why timestamp with timezone mapped to OffsetDatetime instead of Instant or ZonedDateTime

### DISCUSSION THREAD

**Discussion by WestFarmer** [mybatis/generator#1107] [Q&A] [answered]
I was confused by this [SO QUESTION](https://stackoverflow.com/questions/30234594/whats-the-difference-between-zoneddatetime-and-offsetdatetime)

> I'd go the opposite way: the correct one to use is usually either Instant or ZonedDateTime. IMO OffsetDateTime only exist because many existing systems (wrongly) treated "time offset" to be sufficient information about timezones (i.e. they communicate/store/handle just an offset when they should be communicating a time zone name)

I want to use the UTC time to store and transfer all timestamps, but show local time to users  according their timezone.

so In database I choose timestamp without timezone, and I want to it mapped to java.time.instant.

to achieve this, is this enough?

```
<columnOverride column="created_at" javaType="java.time.Instant" jdbcType="timestamp" property="createdAt"></columnOverride>
```

**Comment by jeffgbutler** [comment, marked as answer]
It should work. MyBatis has a built in type handler for java.time.Instant.
